apps/storybook-react/stories/SelectionControls.Checkbox.stories.tsx

```diff
@@ -0,0 +1,211 @@
+import styled from 'styled-components'
+import { action } from '@storybook/addon-actions'
+import { checkbox } from '@equinor/eds-icons'
+import { useForm } from 'react-hook-form'
+import { Meta, Story } from '@storybook/react'
+
+Icon.add({ checkbox })
+
+const BlockCheckbox = styled(Checkbox)`
+  display: flex;
+`
+
+const UnstyledList = styled.ul`
+  margin: 0;
+  padding: 0;
+  list-style-type: none;
+`
+
+export default {
+  title: 'Components/Selection controls/Checkbox',
+  component: Checkbox,
+} as Meta
+
+export const Default: Story<CheckboxProps> = (args) => (
+  <Checkbox label="Play with me" {...args} />
+)
+
+export const SingleCheckbox: Story<CheckboxProps> = () => {
+  // Use this to set the input to indeterminate = true as this must be done via JavaScript
+  // (cannot use an HTML attribute for this)
+  const indeterminateRef = useRef()
+  // State for controlled example
+  const [checked, updateChecked] = useState(false)
+
+  return (
+    <UnstyledList>
+      <li>
+        <Checkbox label="Check me" />
+      </li>
+      <li>
+        <Checkbox label="You can't check me!" disabled />
+      </li>
+      <li>
+        <Checkbox label="I'm preselected" defaultChecked />
+      </li>
+      <li>
+        <Checkbox label="You can't uncheck me!" disabled defaultChecked />
+      </li>
+      <li>
+        <Checkbox
+          label="I'm in indeterminate state"
+          indeterminate
+          ref={indeterminateRef}
+        />
+      </li>
+      <li>
+        <Checkbox
+          label="I'm a controlled component"
+          onChange={(e) => {
+            updateChecked(e.target.checked)
+          }}
+          checked={checked}
+        />
+      </li>
+    </UnstyledList>
+  )
+}
+
+export const GroupedCheckbox: Story<CheckboxProps> = () => {
+  return (
+    <fieldset>
+      <legend>
+        We are in this together!
+        <span role="img" aria-label="raising hands emoji">
+          🙌
+        </span>
+      </legend>
+      <UnstyledList>
+        <li>
+          <Checkbox label="Check me first" name="multiple" value="first" />
+        </li>
+        <li>
+          <Checkbox label="Check me second" name="multiple" value="second" />
+        </li>
+        <li>
+          <Checkbox label="Check me third" name="multiple" value="third" />
+        </li>
+      </UnstyledList>
+    </fieldset>
+  )
+}
+
+type FormData = {
+  favourites: string[]
+  agree: string
+}
+
+export const WithFormsControl: Story<CheckboxProps> = () => {
+  // Example with external forms library, react-hook-form
+  // eslint-disable-next-line @typescript-eslint/unbound-method
+  const { register, handleSubmit, watch, errors } = useForm<FormData>()
+  const [isSubmitted, updateIsSubmitted] = useState(false)
+  const [formData, updateFormData] = useState<FormData>(null)
+
+  const onSubmit = (data: FormData) => {
+    updateFormData(data)
+    updateIsSubmitted(true)
+    action('onSubmit')(data)
+  }
+  console.log(watch('example'))
```

Comment:
🔥  Remove logs

Reply:
It's from the react-hook-library and is supposed to show what's going on inside it. Documented at their page. :)

Reply:
Cool!